mmv1/products/vertexai/Index.yaml

```diff
@@ -131,6 +131,15 @@ properties:
               performed. Exact reordering is a procedure where results returned by an
               approximate search algorithm are reordered via a more expensive distance computation.
               Required if tree-AH algorithm is used.
+          - !ruby/object:Api::Type::String
+            name: 'shardSize'
+            description: |-
+              Index data is split into equal parts to be processed. These are called "shards".
+              The shard size must be specified when creating an index. The value must be one of the followings:
+              * SHARD_SIZE_SMALL: Small (2GB)
+              * SHARD_SIZE_MEDIUM: Medium (20GB)
+              * SHARD_SIZE_LARGE: Large (50GB)
+            default_value: 'SHARD_SIZE_MEDIUM'
```

Comment:
```suggestion
            immutable: true
            default_value: 'SHARD_SIZE_MEDIUM'
```

I tested if the shard_size is updatable by updating an acceptance test. And, it was immutable.


<details><summary>Details</summary>

```console
2023/07/08 22:48:49 [DEBUG] Google API Request Details:
---[ REQUEST ]---------------------------------------
PATCH /v1/projects/xxxxx/locations/us-central1/indexes/2430879471524380672?alt=json&updateMask=metadata.config.shardSize HTTP/1.1
Host: us-central1-aiplatform.googleapis.com
User-Agent: Terraform/1.1.8 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/acc
Content-Length: 447
Content-Type: application/json
Accept-Encoding: gzip

{
 "description": "index for test",
 "displayName": "tf-test-test-index1vhym3pjxc",
 "labels": {
  "foo": "bar"
 },
 "metadata": {
  "config": {
   "algorithmConfig": {
    "bruteForceConfig": null,
    "treeAhConfig": {
     "leafNodeEmbeddingCount": 500,
     "leafNodesToSearchPercent": 7
    }
   },
   "approximateNeighborsCount": 150,
   "dimensions": 2,
   "distanceMeasureType": "DOT_PRODUCT_DISTANCE",
   "featureNormType": "NONE",
   "shardSize": "SHARD_SIZE_MEDIUM"
  },
  "contentsDeltaUri": "gs://tf-test-1vhym3pjxc/contents"
 }
}

-----------------------------------------------------
2023/07/08 22:48:50 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Cross-Origin-Opener-Policy-Report-Only: same-origin; report-to="gfe-default_product_name"
Date: Sun, 09 Jul 2023 05:48:50 GMT
Report-To: {"group":"gfe-default_product_name","max_age":2592000,"endpoints":[{"url":"https://csp.withgoogle.com/csp/report-to/default_product_name"}]}
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "name": "projects/xxxxxxx/locations/us-central1/indexes/2430879471524380672/operations/7449995227340734464",
  "metadata": {
    "@type": "type.googleapis.com/google.cloud.aiplatform.v1.UpdateIndexOperationMetadata",
    "genericMetadata": {
      "createTime": "2023-07-09T05:48:50.105615Z",
      "updateTime": "2023-07-09T05:48:50.105615Z"
    }
  },
  "done": true,
  "response": {
    "@type": "type.googleapis.com/google.cloud.aiplatform.v1.Index",
    "name": "projects/xxxxxxx/locations/us-central1/indexes/2430879471524380672"
  }
}

-----------------------------------------------------
2023/07/08 22:48:50 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Cross-Origin-Opener-Policy-Report-Only: same-origin; report-to="gfe-default_product_name"
Date: Sun, 09 Jul 2023 05:48:50 GMT
Report-To: {"group":"gfe-default_product_name","max_age":2592000,"endpoints":[{"url":"https://csp.withgoogle.com/csp/report-to/default_product_name"}]}
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

{
  "name": "projects/xxxxxx/locations/us-central1/indexes/2430879471524380672",
  "displayName": "tf-test-test-index1vhym3pjxc",
  "description": "index for test",
  "metadataSchemaUri": "gs://google-cloud-aiplatform/schema/matchingengine/metadata/nearest_neighbor_search_1.0.0.yaml",
  "metadata": {
    "config": {
      "dimensions": 2,
      "approximateNeighborsCount": 150,
      "distanceMeasureType": "DOT_PRODUCT_DISTANCE",
      "featureNormType": "NONE",
      "algorithmConfig": {
        "treeAhConfig": {
          "leafNodeEmbeddingCount": "500",
          "leafNodesToSearchPercent": 7
        }
      },
      "shardSize": "SHARD_SIZE_SMALL"
    }
  },
  "etag": "AMEw9yOZ1ZfdIlL0t5oaNxNgRKE0n-eQw7GN4BP2jcKMXiD-s6akHi40Jjtm3j3TsIc=",
  "labels": {
    "foo": "bar"
  },
  "createTime": "2023-07-09T05:10:45.818686Z",
  "updateTime": "2023-07-09T05:48:50.105615Z",
  "indexStats": {
    "vectorsCount": "2",
    "shardsCount": 1
  },
  "indexUpdateMethod": "BATCH_UPDATE"
}
```

</details>